authentication/service/src/main/java/eu/europa/ec/mare/usm/authentication/service/impl/AuthenticationServiceBean.java

```diff
@@ -50,316 +47,287 @@
+            }
+            hashedPassword = stringBuilder.toString();
+        }
+
+        return hashedPassword;
     }
 
-    return ret;
-  }
-
-  private Date handleLoginSuccess(Map<String, Object> userMap, String userName)
-  {
-    LOGGER.debug("handleLoginSuccess(" + userName + ") - (ENTER)");
-    
-    dao.recordLoginSuccess(userName);
-    
-    Date ret = dao.getPasswordExpiry(userName);
-    
-    if (isLDAPEnabled() && userMap != null && !userMap.isEmpty()) {
-    	handleSyncWithLDAP(userMap, userName);
-    } else {
-    	LOGGER.debug("No handle sync with LDAP need");
+    private Date handleLoginSuccess(Map<String, Object> userMap, String userName) {
+        LOGGER.debug("handleLoginSuccess(" + userName + ") - (ENTER)");
+
+        dao.recordLoginSuccess(userName);
+
+        Date passwordExpiry = dao.getPasswordExpiry(userName);
+
+        if (isLDAPEnabled() && userMap != null && !userMap.isEmpty()) {
+            handleSyncWithLDAP(userMap, userName);
+        } else {
+            LOGGER.debug("No handle sync with LDAP need");
+        }
+
+        LOGGER.info("handleLoginSuccess() - (LEAVE)");
+        return passwordExpiry;
     }
-    
-    LOGGER.info("handleLoginSuccess() - (LEAVE)");
-    return ret;
-  }
-  
-  private void handleSyncWithLDAP(Map<String,Object> userMap, String userName) {
-	  int personId = dao.getPersonId(userName);
-	  if (personId == 0) {
-		  dao.createPersonForUser(userMap, userName);
-	  } else {
-		  dao.syncPerson(userMap, personId);
-	  }
-  }
-  
-  private boolean handleLoginFailure(String userName) 
-  {
-    LOGGER.debug("handleLoginFailure(" + userName + ") - (ENTER)");
-    
-    dao.recordLoginFailure(userName);
-    
-    boolean ret = false;
-    
-    Properties props = policyProvider.getProperties(AUTHENTICATION_SUBJECT);
-    int freshold = policyProvider.getIntProperty(props, LOCKOUT_FRESHOLD, 0);
-    int duration =  policyProvider.getIntProperty(props, LOCKOUT_DURATION, 0);
-    if (freshold != 0 && duration != 0) {
-      int failures = dao.getLoginFailures(userName);
-      if (failures >= freshold) {
-        Date lockoutExpiry = new Date (System.currentTimeMillis() + 
-                                       duration * ONE_MINUTE);
-        dao.lockUser(userName, lockoutExpiry);
-      }
+
+    private void handleSyncWithLDAP(Map<String, Object> userMap, String userName) {
+        int personId = dao.getPersonId(userName);
+        if (personId == 0) {
+            dao.createPersonForUser(userMap, userName);
+        } else {
+            dao.syncPerson(userMap, personId);
+        }
+    }
+
+    private void handleLoginFailure(String userName) {
+        LOGGER.debug("handleLoginFailure(" + userName + ") - (ENTER)");
+
+        dao.recordLoginFailure(userName);
+
+        Properties props = policyProvider.getProperties(AUTHENTICATION_SUBJECT);
+        int threshold = policyProvider.getIntProperty(props, LOCKOUT_FRESHOLD, 0);
```

Comment:
lol "freshold"...

Reply:
Would love to change it but I don't know who sets that property so I don't dare touch it.